ui/app/src/components/WidgetDialog/WidgetDialog.tsx

```diff
@@ -50,16 +53,25 @@ export function WidgetDialog(props: WidgetDialogProps) {
     <EnhancedDialog
       title={title}
       maxWidth="xl"
-      classes={{ paper: classes.widgetDialogPaper }}
+      classes={{ ...(fullHeight && { paper: classes.widgetDialogPaper }) }}
       isSubmitting={isSubmitting}
       {...rest}
     >
       <WidgetDialogContext.Provider value={context}>
-        <section className={classes.widgetDialogBody}>
+        <section {...(fullHeight && { className: classes.widgetDialogBody })}>
           <Suspencified>
             <Widget
               {...widget}
-              overrideProps={{ onSubmittingAndOrPendingChange, isSubmitting, mountMode: 'dialog', ...extraProps }}
+              overrideProps={{
+                onSubmittingAndOrPendingChange,
+                isSubmitting,
+                mountMode: 'dialog',
+                ...{
+                  ...extraProps,
+                  ...(extraProps?.onCancel && { onCancel: createCallback(extraProps.onCancel, dispatch) }),
+                  ...(extraProps?.onSuccess && { onCancel: createCallback(extraProps.onSuccess, dispatch) })
```

Comment:
I don't think this isn't the widget dialog's responsibility. Redux-specific interventions should be made by the consumer.